pkg/runtime/reconciler_test.go

```diff
@@ -154,23 +154,22 @@ func TestReconcilerUpdate(t *testing.T) {
 
 	r, kc := reconcilerMocks(rmf)
 
-	statusWriter := &ctrlrtclientmock.StatusWriter{}
-	kc.On("Status").Return(statusWriter)
-	statusWriter.On("Patch", ctx, latestRTObj, client.MergeFrom(desiredRTObj)).Return(nil)
 	kc.On("Patch", ctx, latestRTObj, client.MergeFrom(desiredRTObj)).Return(nil)
 
 	// With the above mocks and below assertions, we check that if we got a
 	// non-error return from `AWSResourceManager.ReadOne()` and the
 	// `AWSResourceDescriptor.Delta()` returned a non-empty Delta, that we end
 	// up calling the AWSResourceManager.Update() call in the Reconciler.Sync()
 	// method,
-	err := r.Sync(ctx, rm, desired)
+	_, err := r.Sync(ctx, rm, desired)
 	require.Nil(err)
 	rm.AssertCalled(t, "ReadOne", ctx, desired)
 	rd.AssertCalled(t, "Delta", desired, latest)
 	rm.AssertCalled(t, "Update", ctx, desired, latest, delta)
+	// No changes to metadata or spec so Patch on the object shouldn't be done
 	kc.AssertNotCalled(t, "Patch", ctx, latestRTObj, client.MergeFrom(desiredRTObj))
-	statusWriter.AssertCalled(t, "Patch", ctx, latestRTObj, client.MergeFrom(desiredRTObj))
+	// Only the HandleReconcilerError wrapper function ever calls patchResourceStatus
```

Comment:
nit: should the test be renamed to TestSync_Update() , which is correct representation of the test.